### DISCUSSION TITLE
MONAI Bundle as an isolated python library

### DISCUSSION THREAD

**Discussion by surajpaib** [Project-MONAI/MONAI#7305] [Q&A]
**Is your feature request related to a problem? Please describe.**
MONAI Bundle can be used to configure a wide range of workflows, and I see that a few issues are open for configuring it in simpler/ non-Pytorch environments.

https://github.com/Project-MONAI/tutorials/issues/1454
https://github.com/Project-MONAI/tutorials/issues/1453

On that note, earlier this year, for a traditional ML project based on [optuna](https://optuna.org/), I attempted to configure my search strategies using monai bundle. However, the issue was that the bundle is tied to MONAI core, which comes with a huge set of dependencies such as pytorch (~700MB), cudnn (~700MB), cublas (~400MB), nccl (~200MB), cuparse (~200MB), etc.

**Describe the solution you'd like**
If I want to use a bundle for other non-pytorch projects, it seems counterintuitive for end-users to download such heavy dependencies. 

Is there a plan to release the bundle as a standalone? 

I personally think this would be amazing as the bundle (with its config parser, resolvers, and CLI overrides) can be a very useful tool for several workflows beyond MONAI-based deep learning. For example, this could be a potential replacement for the more complex [Hydra](https://hydra.cc/docs/intro/) (which we used earlier and abandoned in favor of MONAI bundle).

**Comment by KumoLiu** [comment]
Hi @surajpaib, thanks for your interest here!
It may be hard to decouple the bundle from MONAI. You could specify your own JSON or YAML file.

**Reply by surajpaib** [reply]
Hi @KumoLiu,

We have used the bundle for our Pytorch projects with our YAMLs. The problem is that for other projects, MONAI installs so many dependencies by default, which may be unnecessary.

Are there any ways around this?

**Reply by ericspod** [reply]
The best solution I can think of is to first install the CPU version of Pytorch to avoid the larger packages. Bundles are so tightly coupled with MONAI, which in turn is coupled to Pytorch, it's really not feasible to separate these pieces. We would have to re-engineer the bundle concept to be agnostic to how it's used and there probably isn't any demand for that.
